src/main/java/blockchain/Parameters.java

```diff
@@ -11,7 +11,8 @@
 	private int InitialBalance = 20; // balance to start with at launch
 	private int TxMin = 4; // minimum number of Tx in a block
 	private boolean Mode = true; // honest or malicious
-	
+	private int Token = 20; // tokens to start with at launch
+	private boolean Chain = false; //Mode of lightchain (0 -> original || 1 -> Smartcontract)
```

Comment:
`Chain` has a reserved meaning in blockchain codebases and is not related to what is used here. Suggest to have the `OperationMode` instead. Also, instead of a boolean value, recommend it be a string using the constants:

```
public static final String MODE_NATIVE = "native_mode";
public static final String MODE_CONTRACT = "contract_mode";

private String OperationMode = MODE_NATIVE;
```

```
// somewhere else in the code where you use it (it may need some hammering)
if(Strings.equal(Parameters.OperationMode, Parameters.MODE_NATIVE){
   // do smt
}
```